tui_main.c

```diff
@@ -1228,7 +1228,7 @@ void ged_draw(Ged* a, WINDOW* win) {
 
 void ged_adjust_bpm(Ged* a, Isz delta_bpm) {
   Isz new_bpm = (Isz)a->bpm;
-  if (new_bpm < INT_MAX - delta_bpm)
+  if (delta_bpm < 0 || new_bpm < INT_MAX - delta_bpm)
```

Comment:
I hope I understood you correctly here. Did you mean that `INT_MAX - delta_bpm` in the comparison could lead to an overflow when `delta_bpm` is negative? This additional check should short-circuit the condition before attempting that calculation.

Reply:
Yes. This code would still be susceptible to problems if code elsewhere in `tui_main.c` adjusted the BPM by amounts other than 1, but as it is now, it should be fine.